src/sentry/api/endpoints/organization_sentry_function

```diff
@@ -0,0 +1,31 @@
+from uuid import uuid4
+
+from django.template.defaultfilters import slugify
+from rest_framework import serializers
+from rest_framework.response import Response
+
+from sentry.api.bases import OrganizationEndpoint
+from sentry.api.serializers.rest_framework import CamelSnakeSerializer
+
+
+class SentryFunctionSerializer(CamelSnakeSerializer):
+    name = serializers.CharField()
+    author = serializers.CharField(required=False, allow_blank=True, allow_null=True)
+
+
+class OrganizationSentryFunctionEndpoint(OrganizationEndpoint):
+    # Creating a new sentry function
+    def post(self, request, organization):
+        serializer = SentryFunctionSerializer(data=request.data)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+
+        data = serializer.validated_data
+        data["slug"] = slugify(data["name"])
```

Comment:
Do slugs need to be globally unique or only unique within an organization?

Reply:
For now, the slug should be unique within an organization. May add another field "global_slug", if we want to incorporate a marketplace / allow users to publish functions.

Reply:
Should you have validation to ensure that the slug is unique within the organization?

Reply:
I think that's fair to leave as a TODO for now but yea was should have it